src/XrdCms/XrdCmsCluster.cc

```diff
@@ -1813,51 +1812,113 @@ XrdCmsNode *XrdCmsCluster::SelbyLoad(SMask_t mask, XrdCmsSelector &selR)
+                          }
+                          else if (sp->myLoad > np->myLoad)             sp=np;
+                      }
+                   Multi = true;
+                  }
+               }
           }
-          Multi = true;
-   }
-// pick a random weighed node
-//
-   if ( STHi > 1 ){
-   static std::random_device rand_dev;
-   static std::mt19937 generator(rand_dev());
-   static std::uniform_int_distribution<int> distr(randomSel,selCap);
-   randomSel = distr(generator);
-   for(int i=0;i<=STHi;i++){
-     if(randomSel<=weighed[i]){
-       sp=NodeTab[i];
-       break;
-     }
-   }
-   }
-
-   delete [] weighed;
+    if (useWR){
+    // pick a random weighed node
+    //
+    static std::random_device rand_dev;
+    static std::default_random_engine generator(rand_dev());
+    static std::uniform_int_distribution<int> distr(randomSel,selCap);
+    randomSel = distr(generator);
+    for(int i=0;i<=STHi;i++){
+      if(randomSel<=Weighed[i]){
+        sp=NodeTab[i];
+        break;
+      }
+    }
+    }
 // Check for overloaded node and return result
-// //
+//
    if (!sp) return calcDelay(selR);
-//   RefCount(sp, Multi, selR.needSpace);
+   RefCount(sp, Multi, selR.needSpace);
    return sp;
 }
 
+/******************************************************************************/
+/*                             S e l b y L o a d R                            */
+/******************************************************************************/
+
+// Caller must have the STMutex locked. The returned node, if any, is unlocked.
+
+XrdCmsNode *XrdCmsCluster::SelbyLoadR(SMask_t mask, XrdCmsSelector &selR)
+{
+  static std::random_device rand_dev;
+  static std::default_random_engine generator(rand_dev());
+
+  XrdCmsNode *np = nullptr, *sp = nullptr;
+  bool reqSS = (selR.needSpace & XrdCmsNode::allowsSS) != 0;
+
+  // Scan for a node (preset possible, suspended, overloaded, full, and dead)
+
+  selR.Reset();
+  SelTcnt++;
+int totWeight = 0;
+
+  for (int i = 0; i <= STHi; ++i) {
+    NodeWeight[i] = 0; // make node unselectable first
+
+    if (!((np = NodeTab[i]) && (np->NodeMask & mask)))
+      continue;
+
+    if (!(selR.needNet & np->hasNet)) { selR.xNoNet = true; continue; }
+
+    selR.nPick++;
+
+    if (np->isOffline)                { selR.xOff  = true; continue; }
+    if (np->isBad)                    { selR.xSusp = true; continue; }
+    if (np->myLoad > Config.MaxLoad)  { selR.xOvld = true; continue; }
+
+    if (selR.needSpace) {
+      if (np->DiskFree < np->DiskMinF || (reqSS && np->isNoStage)) {
+        selR.xFull = true;
+        continue;
+      }
+    }
+
+    // If node passes filters, give it a weight
+    totWeight += Config.P_fuzz + (100 - np->myLoad);
+    NodeWeight[i] = totWeight;
+  }
+
+  std::uniform_int_distribution<int> distr(1, totWeight);
+  int selected = distr(generator);
+
+  for (int i = 0; i <= STHi; ++i) {
+    if (NodeWeight[i] < selected)
```

Comment:
comment to explain - selection happens in intervals from previous value to the NodeWeight value